### PR TITLE
MM-42637: fixed `/test embedded create` command

### PR DESCRIPTION
**Pull request by levb**
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-42637

### CODE REVIEW

File: embedded.go [hunk 1]
```diff
@@ -25,9 +32,9 @@ func initHTTPEmbedded(r *mux.Router) {
 }
 
 func handleCreateEmbedded(creq *apps.CallRequest) apps.CallResponse {
-	client := appclient.AsBot(creq.Context)
+	client := appclient.AsActingUser(creq.Context)
```
Comment (mickmister) [inline embedded.go:35]: Does this really fix the issue? Is there an underlying issue with using the bot's token here? The issue occurs even when the bot is a member of the current channel.

Comment (levb) [reply embedded.go:35]: Yes and no, the real issue was `.ChannelID`->`.Channel.Id` change just below. But in my case, I tried it in a channel that the bot was NOT a member of the channel, and that was an error as well.

File: embedded.go [hunk 2]
```diff
@@ -25,9 +32,9 @@ func initHTTPEmbedded(r *mux.Router) {
...
 	p := &model.Post{
-		ChannelId: creq.Context.ChannelID,
+		ChannelId: creq.Context.Channel.Id,
```
Comment (mickmister) [inline embedded.go:37]: Is it possible for us to "hide" the `Context.ChannelID` field that will never be defined?

**Review by mickmister — APPROVED** [review]

File: embedded.go (hunk 2 above)
Comment (hanzei) [reply embedded.go:37]: At some point, yes. But it requires a restructuring: https://github.com/mattermost/mattermost-plugin-apps/blob/4d52c70c63c1e77e83ef7241fe15ff2e3954f54b/apps/context.go#L18

**Comment by levb** [comment]
@DHaussermann a gentle ping? The embedded binding test should work now.

**Review by DHaussermann — APPROVED** [review]
@levb This was functional and I found issues. The only odd behavior was how the debugger in the dev console was behaving. 

I can circle back to this later if this is a problem.

**Pull request merged**